File: DailyChallenge/2020/March/18.java
//https://leetcode.com/explore/challenge/card/march-leetcoding-challenge-2021/590/week-3-march-15th-march-21st/3676/

class Solution {
    public int wiggleMaxLength(int[] nums) {
        if(nums == null || nums.length==0)
            return 0;
        if(nums.length<2){
            return nums.length;
        }

        int count=1;


        for(int i=1, j=0; i<nums.length; j=i, i++){
            if(nums[j]<nums[i]){
                count++;
                while(i<nums.length-1 && nums[i]<=nums[i+1]){
                    i++;
                }
            }else if(nums[j]>nums[i]){
                count++;
                while(i<nums.length-1 && nums[i]>=nums[i+1]){
                    i++;
                }
            }
        }

        return count;
    }
}

File: DailyChallenge/2020/May/18.py
# https://leetcode.com/explore/challenge/card/may-leetcoding-challenge-2021/600/week-3-may-15th-may-21st/3747/

import collections


class Solution(object):
    def findDuplicate(self, paths):
        """
        :type paths: List[str]
        :rtype: List[List[str]]
        """
        files = collections.defaultdict(list)
        for path in paths:
            s = path.split(" ")
            for i in xrange(1,len(s)):
                file_name = s[0] + "/" + s[i][0:s[i].find("(")]
                file_content = s[i][s[i].find("(")+1:s[i].find(")")]
                files[file_content].append(file_name)

        result = []
        for file_content, file_names in files.iteritems():
            if len(file_names)>1:
                result.append(file_names)
        return result
